### ISSUE TITLE
Provide a new option SOURCE_MAP (Boolean) to generate source map

### ISSUE THREAD

**Issue by angelozerr** [javacc21/javacc21#36]
JavaCC21 is able to generate comments in the generated Java code about the line of the JavaCC grammar which have generated the code content:

```java
 // QEL.javacc, line 130
    final public void Expression() throws ParseException {
        if (trace_enabled) LOGGER.info("Entering production defined on line 130 of QEL.javacc");
        if (cancelled) throw new CancellationException();
```

It's very helpful, but for tooling this information is very hard to use it. I mean it should be nice for toolings to provide a navigation feature from the grammar to the generated code.

To fix this problem, it should be very nice to provide a new option `SOURCE_MAP`  to generate source map https://sourcemaps.info/spec.html. This source map is used generally for debugging a language (ex : debug TypeScript uses source map with the generated JavaScript code, the nodejs debugger which is able to debug JavaScript only provides the capability to debug TypeScript files).

Generating source map will give the capability for JavaCC will give the capability for toolings to navigate from the JavaCC grammar to the generated code.

**Comment by revusky** [comment]
Yes, I am aware that there is the potential for quite a bit of improvement on this. I am wondering what is the relationship between the sourcemap you mention and [this](https://parsers.org/t/smap-a-k-a-jsr-45-how-easy-would-it-be-to-apply-to-javacc/29)? I mean JSR 45.

**Comment by revusky** [comment]
I'm closing this as a housekeeping measure. All ongoing development in at https://github.com/congo-cc/congo-parser-generator
